k8s/cmd/admission-webhook/utils.go

```diff
@@ -175,3 +182,56 @@ func readRequest(r *http.Request) ([]byte, error) {
 	}
 	return body, nil
 }
+
+func addVolumeMounts(spec *corev1.PodSpec, added corev1.VolumeMount, basePath string) (patch []patchOperation) {
+	for i := 0; i < len(spec.Containers); i++ {
+		path := basePath + "/containers/" + strconv.Itoa(i) + "/volumeMounts"
+		patch = append(patch, addVolumeMount(added, path, len(spec.Containers[i].VolumeMounts) == 0)...)
+	}
+
+	for i := 0; i < len(spec.InitContainers); i++ {
+		path := basePath + "/initContainers/" + strconv.Itoa(i) + "/volumeMounts"
+		patch = append(patch, addVolumeMount(added, path, len(spec.InitContainers[i].VolumeMounts) == 0)...)
+	}
+
+	return
+}
+
+func addVolumeMount(added corev1.VolumeMount, path string, first bool) (patch []patchOperation) {
+	var value interface{}
+
+	if first {
+		value = []corev1.VolumeMount{added}
+	} else {
+		path = path + "/-"
+		value = added
+	}
+
+	patch = append(patch, patchOperation{
+		Op:    "add",
+		Path:  path,
+		Value: value,
+	})
+
+	return
+}
+
+func addVolume(spec *corev1.PodSpec, added corev1.Volume, basePath string) (patch []patchOperation) {
```

Comment:
[minor] consider to apply basePath after creating patches to reduce parameters in methods here